### PR TITLE
add reanimated v2 support to expo and nextjs

### PR DESCRIPTION
**Pull request by benschac**
add react-native-reanimated to expo go, nextjs

This still needs a bit more polish but just wanted to put up something and get general feedback. Or if it's even something the starter should support out of the box? 

My assumption is that developers would like reanimated configured out of the box, and working with nextjs.

I needed to add web support for reanimated to get rngh working on web for work and thought it would generally be helpful to add to the starter.

I haven't tried it with any of the moti / tamagui animations yet.

@natew - happy to add polish if this is something that'll be added to the starter. If not feel free to close.

CleanShot 2023-05-27 at 23 50 06

### CODE REVIEW

**Comment by benschac** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1223** <a href="https://app.graphite.dev/github/pr/tamagui/tamagui/1223" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/tamagui/tamagui/1223?utm_source=stack-comment).

File: starters/next-expo-solito/apps/expo/package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "react-dom": "^18.2.0",
     "react-native": "^0.71.4",
     "react-native-gesture-handler": "^2.9.0",
+    "react-native-reanimated": "~2.14.4",
```
Comment (benschac) [inline starters/next-expo-solito/apps/expo/package.json:34]: expo go only supports reanimated v2 atm

File: starters/next-expo-solito/packages/app/features/animations/screen.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    <YStack f={1} jc="center" ai="center" p="$4" space>
+      <YStack maw={600}>
+        <Paragraph ta="center">Heres an example reanimated (v2) animation.</Paragraph>
+        <AnimatedStyleUpdateExample />
+      </YStack>
+    </YStack>
+  )
+}
+
+export default function AnimatedStyleUpdateExample(props) {
```
Comment (benschac) [inline starters/next-expo-solito/packages/app/features/animations/screen.tsx:28]: copy-pasta from the expo docs

**Comment by natew** [comment]
I would leave out the example stuff for now we don't want more maintenance burden but the configuration is ok. Works with expo go right?

**Comment by benschac** [comment]
Yerp, need to confirm I didn’t break anything with Tamagui/reanimated or Tamagui/moti. Its a bit slower since nextjs swc plug-in for animated is still WIP and nextjs is falling back to babel/webpack.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

File: starters/next-expo-solito/packages/ui/src/animations.ts [hunk 3]
```diff
@@ -1,4 +1,12 @@
 import { createAnimations } from '@tamagui/animations-react-native'
+// tried using createReanimatedAnimations and kept getting an error with
+// TypeError: Cannot read property 'length' of undefined
+// This error is located at:
+//     in Square (created by TamaguiAnimationDemo)
+//     in TamaguiAnimationDemo (created by HomeScreen)
+//     in RCTView (created by View)
+
+import { createAnimations as createReanimatedAnimations } from '@tamagui/animations-reanimated'
```
Comment (benschac) [inline starters/next-expo-solito/packages/ui/src/animations.ts:9]: I think I could eventually figure this out, assuming you have a lot more context and probably know what's going on @natew 

Assuming tamagui react-native-animations and tamagui react-reanimated packages should be able to be used in a tamagui config `animations` object.

**Comment by benschac** [comment]
handing this off to @natew 


### Unfinished
- I was unable to get `@tamagui/animations-reanimated` to work in the tamagui animation config. Noted in PR
- Unsure about if you want to keep animations here for reference. Also, assuming you'd probably want to style this a bit differently.

### Done
- Reanimated 2 is working on native (expo) and web. Tested on web, ios, android
- `@tamagui/animations-react-native` is working with reanimated 2

**Pull request closed without merging**